packages/tableofcontents/init.lua

```diff
@@ -58,14 +58,19 @@ end
 -- (Similar to SU.contentToString(), but allows passing typeset
 -- objects to functions that need plain strings).
 local function _nodesToText (nodes)
+  local spc = SILE.measurement("0.8spc"):tonumber() -- approx. see below.
```

Comment:
Any chance we could use `1 space - spaceshrinkfactor`? And/or if we are pulling numbers out of a hat I think I would have used the width of a thin space for this, not a regular space.

Reply:
We could. It's still empirical because cannot know easily what settings (incl. fonts) were active when the kerns or glues were absolutized. Of course, most of the time we'd be quite safe: a basic `\section{Header\kern[width=...] stuff}` will be ok (the kern is absolutized it the context of the header font). The edge cases are things such as `\section{Header\font[size=...]{stuff\kern[width=...] stuf}}` -- I don't expect these to often occur, but who knows... E.g. a small kern within a smaller superscript text, etc.

Reply:
Hmm. The real proper way to get a "minimal" space (depending on various settings... we are not _necessarily_ in a context where space shrink/stretch is active, esp. if used e.g. in section titles which might be ragged right...) would be:
```
local sp = SILE.shaper:measureSpace(SILE.font.loadDefaults({}))
local spc = (sp.length - sp.shrink):tonumber()
```
It then takes out the spaceshrinkfactor if the latter is active (= default settings)... But currently gets bigger than 0.8spc due to the 1.2 spaceenlargementfactor we have...

And whatever formula is used, we'd need to apply some margin, to avoid rounding effect (at best) or again when spaces are not variable... In other terms, when comparing "node.width:tonumber() > spc" to decide whether a kern/glue should give a space:
- The earlier implementation was equivalent to `> 0`, i.e. even a small glue or kern would result in a full space...
- The initial PR uses " > 0.8spc" = as the text outputter currently does (without explanation, though) https://github.com/sile-typesetter/sile/blob/c8a1467494b6cbcf3582c56598cb68f55c11df83/outputters/text.lua#L48
- We _could_ make it a bit more clever and use a space with possibly the shrink factor removed as above, but would still need a ratio of it to be safe, or when spaceshrinkfactor is not active... I'm afraid an empirical 0.x is always needed...

Reply:
Yes I think that is better. I don't have a hard evidential use case for why, but at least in my head based on past experiences I think that feels like a more robust heuristic.

Reply:
Done